metomi/isodatetime/data.py

```diff
@@ -365,50 +433,77 @@ def __init__(self, years=0, months=0, weeks=0, days=0,
             (minutes, "minutes", int, float, None),
             (seconds, "seconds", int, float, None)
         )
-        self.years = years
-        self.months = months
-        self.weeks = None
-        self.days = days
+        self._years = years
+        self._months = months
+        self._weeks = None
+        self._days = days
         if weeks is not None:
             if days is None:
-                self.days = CALENDAR.DAYS_IN_WEEK * weeks
+                self._days = CALENDAR.DAYS_IN_WEEK * weeks
             else:
-                self.days += CALENDAR.DAYS_IN_WEEK * weeks
-        self.hours = hours
-        self.minutes = minutes
-        self.seconds = seconds
-        if (not self.years and not self.months and not self.hours and
-                not self.minutes and not self.seconds and
-                weeks and not days):
-            self.weeks = self.days // CALENDAR.DAYS_IN_WEEK
-            self.years, self.months, self.days = (None, None, None)
-            self.hours, self.minutes, self.seconds = (None, None, None)
+                self._days += CALENDAR.DAYS_IN_WEEK * weeks
+        self._hours = hours
+        self._minutes = minutes
+        self._seconds = seconds
+        if (weeks and not years and not months and not days and
+                not hours and not minutes and not seconds):
```

Comment:
Much easier to read with the positive value first, followed by all negations :+1: